docs/file_uploads.md

```diff
@@ -0,0 +1,70 @@
+# File uploads in BCIERS app
+
+Could use some optimization: https://github.com/bcgov/cas-registration/issues/2123
+
+## Frontend
+
+- RJSF supports file with data-urls: https://rjsf-team.github.io/react-jsonschema-form/docs/usage/widgets/#file-widgets
+- `FileWidget`: this started as a copy/paste from RJSF's [FileWidget](https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/core/src/components/widgets/FileWidget.tsx) and @marcelmueller did some styling to match the designs. It now additionally includes a check for max file size, and possibly other stuff including state mgt.
+
+In the rjsf schema:
+
+```ts
+# schema
+statutory_declaration: {
+  type: "string",
+  title: "Statutory Declaration",
+  format: "data-url",
+}
+
+# uiSchema
+statutory_declaration: {
+  "ui:widget": "FileWidget",
+  "ui:options": {
+    filePreview: true,
+    accept: ".pdf",
+  }
+}
+```
+
+## Backend
+
+#### Ninja field validator
+
+The field is declared as a string, and we validate that we can convert it to a file
+
+```python
+...
+  boundary_map: str
+
+  @field_validator("boundary_map")
+  @classmethod
+  def validate_boundary_map(cls, value: str) -> ContentFile:
+    return data_url_to_file(value)
+```
+
+#### Service
+
+```python
+  DocumentService.create_or_replace_operation_document(
```

Comment:
This service is specifically for handling documents related to operations, not sure if that's also what you're doing in reporting? If not, it's still an example of how to set up a service for documents

Reply:
I've used a different service, but the purpose is similar. We'll probably want to refactor this eventually!